src/components/ConfirmedDeletionButton/ConfirmedDeletionButton.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { FC, memo, useState, MouseEvent, useCallback } from 'react';
+
+import { Button } from '@material-ui/core';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';
+import Popover from '@material-ui/core/Popover';
+import useStyles from './styles';
+
+import ActionButton from 'components/ActionButton';
+
+interface ConfirmedDeletionButtonProps {
+  deleteId: string;
+  deleteMethod: (id: string) => void;
+  deleteType?: string;
+  deleteName?: string;
+}
+
+const ConfirmedDeletionButton: FC<ConfirmedDeletionButtonProps> = ({
+  deleteId,
+  deleteMethod,
+  deleteType,
+  deleteName
+}) => {
+  const styles = useStyles();
+  const [open, setOpen] = useState(false);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const displayText =
+    'Are you sure ' +
+    (deleteType && deleteMethod
+      ? 'that you would like to delete the ' + deleteName + ' ' + deleteType + ' ?'
+      : '?');
+
+  const onClickHandler = useCallback((event: MouseEvent<HTMLButtonElement>): void => {
+    setAnchorEl(event.currentTarget);
+    setOpen(true);
+  }, []);
+
+  const onCloseHandler = useCallback(
+    (accept?: boolean): void => {
+      if (accept) deleteMethod(deleteId);
+      setOpen(false);
+      setAnchorEl(null);
+    },
+    [deleteId, deleteMethod]
+  );
+
+  return (
+    <div className={styles.container}>
+      <Button
+        color="secondary"
+        size="small"
+        startIcon={<FontAwesomeIcon icon={faTrashAlt} />}
+        onClick={onClickHandler}
+      >
+        Delete
+      </Button>
+      <Popover
```

Comment:
This popover is mighty beautiful but I think it should close automatically when clicking elsewhere on the screen

Reply:
Good call.